packages/introspect/src/generate.ts

```diff
@@ -261,16 +262,27 @@ function createDomainJsonSchema(
                 fullName: `${domain.schemaName}.${domain.name}`,
             })
         ] = {
-            // TODO: implement domain type
-            type: 'any',
+            type: domainType(domain.innerType),
             description:
                 domain.comment ??
-                `The ${domain.name} enum from the ${domain.schemaName} schema`,
+                `The ${domain.name} domain from the ${domain.schemaName} schema`,
         };
         return acc;
     }, empty);
 }
 
+function domainType(
+    domainInnerType: DomainDetails['innerType'],
+): JSONSchema['type'] {
+    if (domainInnerType === 'pg_catalog.int4') {
```

Comment:
This looks roughly right to me, I guess we should iterate over the "createWellKnownDefs" type to ensure we're supporting all native PG types.

Reply:
Makes sense. On it